test/utils/StringUtilsSpec.scala

```diff
@@ -0,0 +1,23 @@
+package utils
+
+import org.scalatest.freespec.AnyFreeSpec
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.prop.TableDrivenPropertyChecks
+import utils.StringUtils.trimDataEntry
+
+class StringUtilsSpec extends AnyFreeSpec with Matchers with TableDrivenPropertyChecks {
+
+  "trimDataEntry" - {
+    "should remove leading and/or trailing spaces and carriage returns" in {
+      val trimmedString = "test"
+      val untrimmedString = Seq[String](
+        " test",
+        "test  ",
+        "    test ",
+        " \n test ",
+        "test\r "
+      )
+      untrimmedString.forall(trimDataEntry(_) equals trimmedString)
```

Comment:
equals used the java equals. You must use the shouldBe or shouldEqual

For testing a sequence of values, you could use a Table 

val table = Table(("input", "expected"), (" test ", "test), " te \n st", test))